tests/ssl-opt.sh

```diff
@@ -1305,6 +1305,54 @@ requires_config_enabled MBEDTLS_SSL_MAX_FRAGMENT_LENGTH
             -C "found max_fragment_length extension"
 
 requires_config_enabled MBEDTLS_SSL_MAX_FRAGMENT_LENGTH
+run_test    "Max fragment length: enabled, default, larger message" \
+            "$P_SRV debug_level=3" \
+            "$P_CLI debug_level=3 request_size=20000" \
+            0 \
+            -c "Maximum fragment length is 16384" \
+            -s "Maximum fragment length is 16384" \
+            -C "client hello, adding max_fragment_length extension" \
+            -S "found max fragment length extension" \
+            -S "server hello, max_fragment_length extension" \
+            -C "found max_fragment_length extension" \
+            -c "20000 bytes written in 2 fragments" \
+            -s "16384 bytes read" \
+            -s "3616 bytes read"
```

Comment:
Do these tests work with the large request size work if the configuration being tested is larger/smaller than the default 16K? I think this is not the case, for example, the 16384 check will probably fail. Is there any way that we can make the request_size and the checks dependent on the configured fragment length?

Reply:
Yes, the tests (including the pre-existing ones) depend on the `MBEDTLS_SSL_MAX_CONTENT_LEN` being `16384`, which to my understanding might even hold if the fragment length extension is in use: If it's in use because of memory constraints on the system, then the message buffer and hence `MBEDTLS_SSL_MAX_CONTENT_LEN` would be smaller and the tests wouldn't work. If it's in use because of network constraints or because of memory constraints for the peer, the message buffer would probably keep it's normal size.

Reply:
What I mean is that the tests would fail if `MBEDTLS_SSL_MAX_CONTENT_LEN` was not 16K. Did the previous tests fail in that case as well?

Reply:
Yes, sorry if I wasn't clear: The pre-existing (and untouched) tests also fail when `MBEDTLS_SSL_MAX_CONTENT_LEN` is not `16384`, see all the occurences of `16384` in https://github.com/ARMmbed/mbedtls/blob/development/tests/ssl-opt.sh. 

I remember there was a similar issue lately about tests being specifically tailored at default values of configuration parameters that the user was allowed to change, namely the value of `MAX_INTERMEDIATE_CA`. We could follow a similar strategy to https://github.com/ARMmbed/mbedtls/commit/bcfa6f42e3da922d13722fe912dcb772649e501c to gracefully fail if `MBEDTLS_SSLMAX_CONTENT_LEN` deviates from its default value. What do you think?

Reply:
I am not completely satisfied with the  check in commit bcfa6f4 as it makes assumptions instead of actually extracting the config value from the source code. However, I think its a good short term idea to avoid failures in the testing that we might then need to spend time investigating.

Reply:
I've added the check.